nacossync-worker/src/main/java/com/alibaba/nacossync/timer/FastSyncHelper.java

```diff
@@ -0,0 +1,196 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacossync.timer;
+
+import com.alibaba.nacos.client.naming.utils.CollectionUtils;
+import com.alibaba.nacossync.cache.SkyWalkerCacheServices;
+import com.alibaba.nacossync.constant.MetricsStatisticsType;
+import com.alibaba.nacossync.extension.SyncManagerService;
+import com.alibaba.nacossync.extension.impl.NacosSyncToNacosServiceImpl;
+import com.alibaba.nacossync.monitor.MetricsManager;
+import com.alibaba.nacossync.pojo.model.TaskDO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+
+import static com.alibaba.nacossync.constant.SkyWalkerConstants.MAX_THREAD_NUM;
+
+/**
+ * multi-threaded synchronization Task DO.
+ * @ClassName: FastSyncHelper
+ * @Author: ChenHao26
+ * @Date: 2022/7/19 17:02
+ * @Description:
+ */
+@Service
+@Slf4j
+public class FastSyncHelper {
+    
+    @Autowired
+    private SkyWalkerCacheServices skyWalkerCacheServices;
+    
+    @Autowired
+    private MetricsManager metricsManager;
+    
+    @Autowired
+    private SyncManagerService syncManagerService;
+    
+    @Autowired
+    private NacosSyncToNacosServiceImpl nacosSyncToNacosService;
+    
+    /**
+     * every 200 services start a thread to perform synchronization.
+     * @param taskDOS task list
+     */
+    public void syncWithThread(List<TaskDO> taskDOS) {
+        long startTime = System.currentTimeMillis();
+        List<List<TaskDO>> taskGroupList = averageAssign(taskDOS, MAX_THREAD_NUM);
+        CountDownLatch countDownLatch = new CountDownLatch(taskGroupList.size());
+    
+        // 创建分组线程，每个线程执行自己的任务
+        for (int i = 0; i < taskGroupList.size(); i++) {
+            new SyncThread(i, countDownLatch, taskGroupList.get(i)).start();
+        }
+        try {
+            countDownLatch.await();
+        }catch (InterruptedException exception) {
+            exception.printStackTrace();
+        }
+        
+        log.info("新增同步任务数量 {}, 执行耗时：{}ms",taskDOS.size() , System.currentTimeMillis() - startTime);
+    }
+    
+    /**
+     * every 200 services start a thread to perform synchronization.
+     * @param taskDO task info
+     * @param filterServices filterServices
+     */
+    public  void syncWithThread(TaskDO taskDO, List<String> filterServices) {
+        long start=System.currentTimeMillis();
+        List<List<String>> lists = averageAssign(filterServices, MAX_THREAD_NUM);
+        CountDownLatch countDownLatch = new CountDownLatch(lists.size());
+        
+        //创建分组线程，每个线程执行自己的任务
+        for (int i = 0; i < lists.size(); i++) {
+            new SyncThread(i, countDownLatch, lists.get(i), taskDO).start();
```

Comment:
plz, consider if need to use a thread pool.

Reply:
> 

I will consider it